core/src/main/scala/pickling/FastTags.scala

```diff
@@ -3,18 +3,64 @@ package scala.pickling
 import scala.language.experimental.macros
 
 import scala.pickling.internal._
+import scala.reflect.api.Mirror
 
 import scala.reflect.runtime.{universe => ru}
 import scala.reflect.ClassTag
 
+/**
+ * A "tag" denoting information about a runtime type.
+ * This tag is meant to be extermely efficient for simple runtime checks, avoiding a full reflection overhead, while
+ * also *allowing* full runtime checks.
+ *
+ * Notes:
+ *
+ * 1. Currently the equals method does "stringy" comparison of types.  For a full equality comparison, you'll need to
+ *    fully reify the Type (tpe).
+ * 2. Calling `mirror` or `tpe` may cause runtime reflection to be used.
+ *
+ *
+ * @tparam T
+ */
 trait FastTypeTag[T] extends Equals {
+  /** @return The mirror known to be in use when this FastTypeTag was created.
+    *
+    * It is not guaranteed to be the correct mirror to reify the type against.
+    */
   def mirror: ru.Mirror
+
+  /**
+   * @return The full Type of T.   This method may need to use runtime reflection to reconstruct the full type.
+   */
   def tpe: ru.Type
+
+  /** A stringified key that can be used to denote this type.   This key should be unique for types within scala,
+    * although the key will *not* determine uniqueness between types loaded on different classloaders.
+    *
+    * @return  A stringy type key.
+    */
   def key: String
+
+  /**
+   * @param otherMirror The mirror where we should reconsititute the Type inside.
+   * @return  A new Type instance that has reconstructed the full Type.
+   */
+  def reflectType(otherMirror: ru.Mirror): ru.Type = typeFromString(otherMirror, key)
+
+  /**
+   * Tests whether this tag is effectively a primitive type.  Note: We duplicate logic
+   * out of regular runtime reflection here to avoid the burden of requiring runtime reflection.
+   */
+  def isEffectivelyPrimitive: Boolean =
+    FastTypeTag.EffectivePrimitiveTags.contains(key)
```

Comment:
Good idea. There's really no need to rely on runtime reflection for this. Much better this way.